woocommerce-payments.php

```diff
@@ -150,6 +144,11 @@ function () {
 function wcpay_init() {
 	require_once WCPAY_ABSPATH . '/includes/class-wc-payments.php';
 	WC_Payments::init();
+	/**
+	 * Needs to be loaded as soon as possible
+	 * Check https://github.com/Automattic/woocommerce-payments/issues/4759
+	 */
+	\WCPay\WooPay\WooPay_Session::init();
```

Comment:
I'm a little bit suspicious on whether this will work or not. Before we moved the `init` call to the top of the file it was effectively called in `WC_Payments::init()`, so moving it here actually puts it _later_ in the execution cycle 🤔

https://github.com/Automattic/woocommerce-payments/blob/4b78576848769d30aadbe0c68ac100828c681706/includes/class-wc-payments.php#L337

(`maybe_register_platform_hooks` was the call that registered the action before https://github.com/Automattic/woocommerce-payments/pull/4762/)

I wonder if maybe we've changed how things actually work significantly enough for this no longer to be an issue? I'll make sure to test this part as thoroughly as I can 👍

Reply:
Yeah I was wondering the same about if things have changed enough for this to not be an issue, but it is a concern of mine. In my initial testing it didn't appear to be an issue, but we may have to go the route of decoupling any reliance on `WC_Payments` being initialized.

Reply:
I've not reverted this part of the PR and instead removed the `WC_Payments` dependency from the `is_woopay_enabled` check.

Reply:
Awesome, thank you Brian! 🙌